client_test.go

```diff
@@ -107,6 +108,90 @@ func TestBaseClient_Create(t *testing.T) {
 	})
 }
 
+func TestBaseClient_SearchWithContext_ValidQuery_ReturnsExpectedResult(t *testing.T) {
```

Comment:
can you create 1 test for the `SearchWithContext` function, with nested tests? Makes it more organized i.m.o.